src/java/org/apache/cassandra/index/sai/disk/vector/CompactionGraph.java

```diff
@@ -371,9 +419,17 @@ public long addGraphNode(InsertionResult result)
 
     public SegmentMetadata.ComponentMetadataMap flush() throws IOException
     {
-        // header is required to write the postings, but we need to recreate the writer after that with an accurate OrdinalMapper
-        writer.writeHeader(builder.getGraph().getView());
-        writer.close();
+        if (writer != null)
+        {
+            // header is required to write the postings, but we need to recreate the writer after that with an accurate OrdinalMapper
+            writer.writeHeader(builder.getGraph().getView());
+            writer.close();
+        }
+        else
+        {
+            // Close the temporary file so the reader will know it is the end of the file.
+            vectorsByOrdinalBufferedWriter.close();
+        }
```

Comment:
Nit: Not sure if I like the fact we have two different kinds of writers and they need different kind of care. But maybe it cannot be better, I don't know. ;)

Also what if we have both `writer` and `vectorsByOrdinalBufferedWriter`? Or is it not possible? Maybe add assert?